streampark-flink/streampark-flink-proxy/src/main/scala/org/apache/streampark/flink/proxy/FlinkShimsProxy.scala

```diff
@@ -81,59 +83,106 @@ object FlinkShimsProxy extends Logger {
+  def getVerifySqlLibClassLoader(flinkVersion: FlinkVersion): ClassLoader = {
+    logInfo(s"add  verify sql lib,flink version:  $flinkVersion")
+    VERIFY_SQL_CLASS_LOADER_CACHE.getOrElseUpdate(s"${flinkVersion.fullVersion}", {
+      val getFlinkTable: File => Boolean = _.getName.startsWith("flink-table")
+      // 1) flink/lib/flink-table*
+      val libTableURL = getFlinkHomeLib(flinkVersion.flinkHome, "lib", getFlinkTable)
+
+      // 2) After version 1.15 need add flink/opt/flink-table*
+      val optTableURL = getFlinkHomeLib(flinkVersion.flinkHome, "opt", getFlinkTable)
+      val shimsUrls = ListBuffer[URL](libTableURL ++ optTableURL: _*)
+
+      // 3) add only streampark shims jar
+      addShimsUrls(flinkVersion, shimsUrls, _.getName.startsWith("streampark-flink-shims"))
+      new ChildFirstClassLoader(
+        shimsUrls.toArray,
+        Thread.currentThread().getContextClassLoader,
+        getFlinkShimsResourcePattern(flinkVersion.majorVersion)
+      )
+    })
+  }
+
+  def addShimsUrls(flinkVersion: FlinkVersion, shimsUrls: ListBuffer[URL], filterJar: File => Boolean): Unit = {
+    val appHome = System.getProperty(ConfigConst.KEY_APP_HOME)
+    require(appHome != null, String.format("%s is not found on System env.", ConfigConst.KEY_APP_HOME))
+
+    val libPath = new File(s"$appHome/lib")
+    require(libPath.exists())
     val majorVersion = flinkVersion.majorVersion
     val scalaVersion = flinkVersion.scalaVersion
-    logInfo(flinkVersion.toString)
+    val streamParkMatcher = getStreamParkLibPattern(scalaVersion)
+
+    def addShimUrl(jar: File): Unit = {
+      if (filterJar.apply(jar)) {
+        shimsUrls += jar.toURI.toURL
+      }
+    }
+
+    libPath.listFiles().foreach((jar: File) => {
+      try {
+        val shimsMatcher = SHIMS_PATTERN.matcher(jar.getName)
+        if (shimsMatcher.matches()) {
+          if (majorVersion == shimsMatcher.group(1) && scalaVersion == shimsMatcher.group(2)) {
+            addShimUrl(jar)
+          }
+        } else {
+          if (INCLUDE_PATTERN.matcher(jar.getName).matches()) {
+            addShimUrl(jar)
+            logInfo(s"include jar lib: ${jar.getName}")
+          }
+          if (streamParkMatcher.matcher(jar.getName).matches()) {
+            addShimUrl(jar)
+            logInfo(s"include streampark lib: ${jar.getName}")
+          }
+        }
+      } catch {
+        case e: Exception => e.printStackTrace()
+      }
+    })
+  }
+
+  /**
+   * Get ClassLoader to verify sql
+   *
+   * @param flinkVersion flinkVersion
+   * @param func         execute function
+   * @tparam T
+   * @return
+   */
+  def proxyVerifySql[T](flinkVersion: FlinkVersion, func: JavaFunc[ClassLoader, T]): T = {
+    val shimsClassLoader = getVerifySqlLibClassLoader(flinkVersion)
+    ClassLoaderUtils.runAsClassLoader[T](shimsClassLoader, new Supplier[T]() {
+      override def get(): T = func.apply(shimsClassLoader)
+    })
```

Comment:
```suggestion
    ClassLoaderUtils.runAsClassLoader[T](shimsClassLoader, () => func(shimsClassLoader))
```

This can be simplify.